src/amqp10_client_connection.erl

```diff
@@ -399,15 +401,17 @@ send_heartbeat(#state{socket = Socket}) ->
     Frame = amqp10_binary_generator:build_heartbeat_frame(),
     socket_send(Socket, Frame).
 
+-dialyzer({no_fail_call, socket_send/2}).
```

Comment:
Looking through various headers, it seems to me that when a socket it created, it's as type `any()`, but `ssl:send/2` and related functions expect a socket that is an `#sslsocket` record. I presume this is an artifact of the evolution of the ssl library, so I have suppressed this and similar warnings.

Reply:
@kjnilsson ^^